com.liferay.blade.eclipse.provider/src/com/liferay/blade/eclipse/provider/JavaFileJDT.java

```diff
@@ -592,4 +594,57 @@ protected File getFile() {
 		return null;
 	}
 
+	private boolean typeMatch(String expectType, String paramType) {
+		boolean match = false;
+
+		if (expectType.endsWith(paramType) || paramType.endsWith(expectType)) {
+			match = true;
+		}
+		else if (expectType.equals("Object") || expectType.equals("java.lang.Object") || expectType.equals("T")
+				|| expectType.equals("E") || expectType.equals("U")) {
+			match = true;
+		}
+		else if (expectType.equals("long")) {
+			if (paramType.equals("long") || paramType.equals("Long") || paramType.equals("java.lang.Long")
+					|| paramType.equals("int") || paramType.equals("short") || paramType.equals("btye"))
+				match = true;
+		}
+		else if (expectType.equals("Long") || expectType.equals("java.lang.Long")) {
+			if (paramType.equals("long") || paramType.equals("Long") || paramType.equals("java.lang.Long")) {
+				match = true;
+			}
+		}
+		else if (expectType.equals("int")) {
+			if (paramType.equals("int") || paramType.equals("Integer") || paramType.equals("java.lang.Integer")
+					|| paramType.equals("short") || paramType.equals("btye"))
+				match = true;
+		}
+		else if (expectType.equals("Integer") || expectType.equals("java.lang.Integer")) {
+			if (paramType.equals("int") || paramType.equals("Integer") || paramType.equals("java.lang.Integer")) {
+				match = true;
+			}
+		}
+		else if (expectType.equals("short")) {
+			if (paramType.equals("short") || paramType.equals("Short") || paramType.equals("java.lang.Short")
+					|| paramType.equals("btye"))
+				match = true;
+		}
+		else if (expectType.equals("Short") || expectType.equals("java.lang.Short")) {
+			if (paramType.equals("short") || paramType.equals("Short") || paramType.equals("java.lang.Short")) {
+				match = true;
+			}
+		}
+		else if (expectType.equals("btye")) {
+			if (paramType.equals("btye") || paramType.equals("Btye") || paramType.equals("java.lang.Btye"))
+				match = true;
+		}
+		else if (expectType.equals("Btye") || expectType.equals("java.lang.Btye")) {
+			if (paramType.equals("btype") || paramType.equals("Btye") || paramType.equals("java.lang.Btye")) {
```

Comment:
Here you check for paramType = 'java.lang.Byte' but on line 619 you don't.  If one is needed here, the other should be too right?  

Also we need more test coverage, we need some specific java classes showing the problems and showing where these new code blocks match it.